src/WordPress/WordPressAPI.php

```diff
@@ -86,23 +129,40 @@ private function formatDomain($domainName)
         return $formattedDomain;
     }
 
+    public function setWordPressClientAPI($wordPressClientAPI)
+    {
+        return $wordPressClientAPI;
+    }
+
     /**
-     * @param null $userId
-     *
      * @return mixed
      */
-    public function getDomainList($userId = null)
+    public function isSubdomainValidCloudflareDomain($wordpressIntegration, $domainName)
     {
-        $domainName = $_SERVER['SERVER_NAME'];
+        $this->wordPressClientAPI = $this->setWordPressClientAPI(new \CF\WordPress\WordPressClientAPI($wordpressIntegration));
+        $response = $this->wordPressClientAPI->getZones();
+        if ($this->wordPressClientAPI->responseOk($response)) {
+            foreach ($response['result'] as $zone) {
+                if (Utils::isSubdomainOf($domainName, $zone['name'])) {
+                    return true;
+                }
+            }
+        }
 
-        return array($this->formatDomain($domainName));
+        return false;
     }
 
-    /**
-     * @return mixed
-     */
-    public function getUserId()
+    public function cacheWordPressDomainName($wordpressIntegration)
```

Comment:
Do you need the whole `$wordpressIntegration`?  Why not take specifically what you need and pass it in through the class's constructor?